### PR TITLE
Fixing a few Things

### PR DESCRIPTION
**Pull request by GamefreakDE**
- Fixed Purple Bandanas
- Changed Scoreboard to always have only one Usergroup
- Made overridetargetid less overriding
- Changed Scaledamage to return true
- If player didnt get his weapons for whatever reason, he will get them

### CODE REVIEW

File: lua/cl_spectator_deathmatch.lua [hunk 1]
```diff
@@ -115,75 +115,27 @@ end
...
+				return GROUP_TERROR
 			end
+		elseif p:IsGhost() and (LocalPlayer():IsSpec() or LocalPlayer():IsGhost()) then
+			return GROUP_DEATHMATCH
 		end
-		return old_vguiRegister(className, tbl, base)
 	end
-	
+	hook.Add("TTTScoreGroup", "TTTScoreGroup_Ghost", ScoreGroupDM)
+	hook.Add("PostGamemodeLoaded", "PostGamemodeLoaded_Ghost", function() AddScoreGroup("DEATHMATCH") end)
```
Comment (markusmarkusz) [inline lua/cl_spectator_deathmatch.lua]: Why?

File: lua/cl_spectator_deathmatch.lua [hunk 2]
```diff
@@ -115,75 +115,27 @@ end
 timer.Create( "SpecDMHeartbeat", 5, 0, SendHeartbeat )
 
 if not SpecDM.IsScoreboardCustom then
+	hook.Add("TTTScoreGroups", "TTTScoreGroups_Ghost", function(parent_panel, player_group_panels)
```
Comment (markusmarkusz) [inline lua/cl_spectator_deathmatch.lua]: Why? Either remove the whole scoreboard code or fix it.

File: lua/cl_spectator_deathmatch.lua (hunk 1 above)
Comment (GamefreakDE) [inline lua/cl_spectator_deathmatch.lua]: If AddScoreGroup gets called earlier its nil due to the function not being loaded.

File: lua/cl_spectator_deathmatch.lua [hunk 3]
```diff
@@ -115,75 +115,27 @@ end
...
-					local pnl = self.rows[ply]
-					if ValidPanel(pnl) then
-						pnl:Remove()
-					end
-					self.rows[ply] = nil
-				end
-				self.rowcount = table.Count(self.rows)
-				self:UpdateSortCache()
-				self:InvalidateLayout()
+	function ScoreGroupDM(p)
```
Comment (markusmarkusz) [inline lua/cl_spectator_deathmatch.lua]: Make this function local. Not global.

File: lua/cl_spectator_deathmatch.lua (hunk 2 above)
Comment (GamefreakDE) [inline lua/cl_spectator_deathmatch.lua]: Whats wrong with the new code?

File: lua/cl_spectator_deathmatch.lua (hunk 3 above)
Comment (GamefreakDE) [inline lua/cl_spectator_deathmatch.lua]: Okay, thats fair enough

File: lua/sv_specdm_overrides.lua [hunk 4]
```diff
@@ -12,7 +12,7 @@ concommand.Add = function(command, func, help)
 end
 
 hook.Add("PlayerTraceAttack", "PlayerTraceAttack_SpecDM", function(ply, dmginfo, dir, trace)
```
Comment (markusmarkusz) [inline lua/sv_specdm_overrides.lua]: Why does this hook even exist?

File: lua/weapons/weapon_ghost_crowbar.lua [hunk 5]
```diff
@@ -118,7 +118,7 @@ function SWEP:OpenEnt(hitEnt)
    end
 end
 
-if SERVER then
+if CLIENT then
```
Comment (markusmarkusz) [inline lua/weapons/weapon_ghost_crowbar.lua]: Remove it completly.

File: lua/cl_spectator_deathmatch.lua (hunk 2 above)
Comment (markusmarkusz) [inline lua/cl_spectator_deathmatch.lua]: It doesn't do anything.
Because no one will be in this Scoregroup.

**Comment by Tommy228** [comment]
> Changed Scoreboard to always have only one Usergroup
Why do you want to do that?

**Comment by GamefreakDE** [comment]
My 1. Problem was that I dont like that fact that you were displayed in the score board twice, and 
2. was that I dont know if it would be even possible to do that again without overriding the function. I am closing this PR at least for now due to these things, but I just hate to override functions if there are hooks that can do that.

**Comment by MinIsMin** [comment]
Can I fix the purple bandanas?

**Pull request closed without merging**